### PR TITLE
doc: how to unset Clear Linux' CFLAGS and CXXFLAGS

### PR DESCRIPTION
**Pull request by marc-hb**
Clear Linux exports to all users a list of "aggressive" compiler and
linker flags. Zephyr's CMake build system will either warn or fail
because of these. Add one magic command that solves the issue.

More background information at https://superuser.com/a/1452523/111302

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

### CODE REVIEW

File: doc/getting_started/installation_linux.rst [hunk 1]
```diff
@@ -85,6 +85,24 @@ Clear Linux:
    sudo swupd bundle-add c-basic dev-utils dfu-util dtc \
      os-core-dev python-basic python3-basic python3-tcl
 
+Clear Linux' focus is *native* performance and security and not
```
Comment (dbkinder) [inline doc/getting_started/installation_linux.rst]: We can avoid the awkward possessive with
```suggestion
The Clear Linux focus is on *native* performance and security and not
```

File: doc/getting_started/installation_linux.rst [hunk 2]
```diff
@@ -85,6 +85,24 @@ Clear Linux:
...
+cross-compilation. For that reason it uniquely exports by default to the
+:ref:`environment <env_vars>` of all users a list of compiler and linker
+flags. Zephyr's CMake build system will either warn or fail because of
+these. To clear the C/C++ flags among these and fix the Zephyr build run
```
Comment (dbkinder) [inline doc/getting_started/installation_linux.rst]:
```suggestion
these. To clear the C/C++ flags among these and fix the Zephyr build, run
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/getting_started/installation_linux.rst [hunk 3]
```diff
@@ -85,6 +85,24 @@ Clear Linux:
...
+The Clear Linux focus is on *native* performance and security and not
+cross-compilation. For that reason it uniquely exports by default to the
+:ref:`environment <env_vars>` of all users a list of compiler and linker
+flags. Zephyr's CMake build system will either warn or fail because of
+these. To clear the C/C++ flags among these and fix the Zephyr build, run
+the following command as root then log out and back in:
+
+.. code-block:: console
+
+   # echo 'unset CFLAGS CXXFLAGS' >> /etc/profile.d/unset_cflags.sh
```
Comment (jukkar) [inline doc/getting_started/installation_linux.rst]: Can't we add this to `~/.zephyrrc` file instead?

Comment (marc-hb) [reply doc/getting_started/installation_linux.rst]: Yes if you find a reliable way to source it (or zephyr-env.sh) that doesn't get overriden. So this doesn't change the problem, it just moves it somewhere else.
I'm already linking to https://docs.zephyrproject.org/latest/guides/env_vars.html which is the best place to discuss all environment questions once for all; however I could insist a bit more on that link, would you like me to?

**Review by dbkinder — APPROVED** [review]
LGTM, thanks!

**Review by nashif — APPROVED** [review]

**Comment by marc-hb** [comment]
Thanks @nashif . Do maintainers have a "squash and merge" option in this repo? https://help.github.com/en/articles/about-pull-request-merges
@dbkinder 's fixes have been merged as a separate commit.

**Comment by nashif** [comment]
we do not have squash/merge, we just rebase and merge, this PR should not have passed CI, not sure what happened. 
When you apply someone's suggestions you should always rebase and squash their changes into the your original commits, should not leave them as is. 

I missed all of this because I did not look at the commits and trusted CI, something definitely failed here.

**Comment by nashif** [comment]
@dbkinder please refrain from suggesting changes until we know what happened here and why it passed CI.

**Comment by marc-hb** [comment]
Thanks @nashif , curious which CI check should have failed? gitlint? I know @ulfalizer who contributed the most to check_compliance.py is a big fan of checking all commits at once, he said so a few times.

**Comment by marc-hb** [comment]
Wait: just remembered CI had been hanging the whole day ("waiting to be reported...") before you merged it. Was it not hung anymore when you merged it?

BTW why do the CI results always disappear after merge? Can't think of any good reason.

**Pull request merged**